policies.go

```diff
@@ -194,6 +209,69 @@ func getExponentialTime(min time.Duration, max time.Duration, attempts int) time
 	return time.Duration(napDuration)
 }
 
+func (e *ExponentialBackoffRetryPolicy) GetRetryType(err error) RetryType {
+	return RetryNextHost
+}
+
+// DowngradingConsistencyRetryPolicy: Next retry will be with the next consistency level
+// provided in the slice
+/*
+* On a read timeout: the operation is retried with the next provided consistency
+level.
+* On a write timeout: if the operation is an :attr:`~.UNLOGGED_BATCH`
+and at least one replica acknowledged the write, the operation is
+retried with the next consistency level.  Furthermore, for other
+write types, if at least one replica acknowledged the write, the
+timeout is ignored.
+* On an unavailable exception: if at least one replica is alive, the
+operation is retried with the next provided consistency level.
+*/
+type DowngradingConsistencyRetryPolicy struct {
+	ConsistencyLevelsToTry []Consistency
+}
+
+func (d *DowngradingConsistencyRetryPolicy) Attempt(q RetryableQuery) bool {
+	currentAttempt := q.Attempts()
+
+	if currentAttempt > len(d.ConsistencyLevelsToTry) {
+		return false
+	}
+	if currentAttempt > 0 {
+		q.SetConsistency(d.ConsistencyLevelsToTry[currentAttempt-1])
+		if gocqlDebug {
+			Logger.Printf("%T: set consistency to %q\n",
+				d,
+				d.ConsistencyLevelsToTry[currentAttempt-1])
+		}
+	}
+	return true
+}
+
+func (d *DowngradingConsistencyRetryPolicy) GetRetryType(err error) RetryType {
+	switch err.(type) {
+	case *RequestErrUnavailable:
+		if err.(*RequestErrUnavailable).Alive > 0 {
```

Comment:
this can be written like

```go
switch v := err.(type) {
case *RequestErrUnavailable:
    // v is now *RequestErrUnavailable
```

See https://golang.org/doc/effective_go.html#type_switch